tests/integration/internal/constants.go

```diff
@@ -392,3 +412,26 @@ func InitializeConstants() error {
 	log.Printf("Supported kind images: %v", KindImages.Supported)
 	return nil
 }
+
+func getKubernetesVersion(
+	t *testing.T,
+) string {
+	v, err := k8s.GetKubernetesClusterVersionE(t)
+	require.NoError(t, err)
+	return v
+}
+
+func GetVersionDependentMetrics(t *testing.T) []string {
+	res := []string{}
+	currVersion := getKubernetesVersion(t)
+
+	for version, ms := range versionDependentMetrics {
+		if currVersion < version {
```

Comment:
This is a string comparison, right? If we want to compare this way, we should use a more structured data type for this.

Reply:
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/version looks like a good choice.